projects/bootloader/src/x86/jump_to_application.c

```diff
@@ -0,0 +1,5 @@
+#include "jump_to_application.h"
+
+void jump_to_application(void) {
+  // unless we try simulating the bootloader, this doesn't have much meaning on x86: do nothing
```

Comment:
Maybe we could add a LOG_DEBUG here to warn someone if they try and run the bootloader on x86

Reply:
Actually,  I'm wondering if we even want to keep this? Would there be a point of allowing people to specify that they want PLATFORM=x86 for the stm32 CAN bootloader? Maybe we could just remove this functionality? For our other code it makes sense for testing though I'm not sure if its as applicable to have an x86 version of CAN bootloader.

Reply:
I added the x86 implementation because the stm32-specific version doesn't compile on x86, so I had to put it in an stm32f0xx directory. Then if we don't have the x86 version we get a linker error from the `jump_to_application()` call in main.c, and we can't put main.c itself in stm32f0xx/ since then we get an linker error complaining about a missing `main`. So it looks like the x86 version is here to stay.

I guess we could limit the bootloader in rules.mk or somewhere to compile only on stm32, but there are parts of the bootloader we'll want to unit test on x86 (e.g. the protocol implementation), so I think this way is good. The `LOG_DEBUG` is a good idea though.